google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/test_helpers/env/CloudEnv.java

```diff
@@ -153,4 +193,57 @@ private static String getRequiredProperty(String prop) {
     }
     return value;
   }
+
+  /**
+   * Captures the request attributes "Grpc.TRANSPORT_ATTR_REMOTE_ADDR" when connection is
+   * established and verifies if the remote address is a DirectPath address. This is only used for
+   * DirectPath testing. {@link ClientCall#getAttributes()}
+   */
+  private ClientInterceptor directPathAddressCheckInterceptor() {
+    return new ClientInterceptor() {
+      @Override
+      public <ReqT, RespT> ClientCall<ReqT, RespT> interceptCall(
+          MethodDescriptor<ReqT, RespT> method, CallOptions callOptions, Channel next) {
+        final ClientCall<ReqT, RespT> clientCall = next.newCall(method, callOptions);
+        return new SimpleForwardingClientCall<ReqT, RespT>(clientCall) {
+          @Override
+          public void start(Listener<RespT> responseListener, Metadata headers) {
+            super.start(
+                new SimpleForwardingClientCallListener<RespT>(responseListener) {
+                  @Override
+                  public void onHeaders(Metadata headers) {
+                    // Check peer IP after connection is established.
+                    SocketAddress remoteAddr =
+                        clientCall.getAttributes().get(Grpc.TRANSPORT_ATTR_REMOTE_ADDR);
+                    if (!verifyRemoteAddress(remoteAddr)) {
+                      throw new RuntimeException(
+                          String.format(
+                              "Synthetically aborting the current request because it did not adhere"
+                                  + " to the test environment's requirement for DirectPath."
+                                  + " Expected test to access DirectPath via %s,"
+                                  + " but RPC was destined for %s",
+                              remoteAddr.toString(), isDirectPathIpv4() ? "ipv4" : "ipv6"));
```

Comment:
I think your parameters are backwards:
`Expected test to access DirectPath via %s, but RPC was destined for %s"`
so the values should be flipped:
`isDirectPathIpv4() ? "ipv4" : "ipv6", remoteAddr.toString()`

Reply:
oops :) Done.